de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/service/impl/BPartnerDAO.java

```diff
@@ -815,14 +815,19 @@ private static final String getBPartnerDiscountSchemaColumnNameOrNull(final BPar
 	@Override
 	public BPartnerLocationId getBilltoDefaultLocationIdByBpartnerId(@NonNull final BPartnerId bpartnerId)
 	{
-		return retrieveBPartnerLocations(bpartnerId)
+		return retrieveBPartnerLocationsInTrx(bpartnerId)
 				.stream()
 				.filter(I_C_BPartner_Location::isBillToDefault)
 				.findFirst()
 				.map(bpLocation -> BPartnerLocationId.ofRepoId(BPartnerId.ofRepoId(bpLocation.getC_BPartner_ID()), bpLocation.getC_BPartner_Location_ID()))
 				.orElse(null);
 	}
 
+	public List<I_C_BPartner_Location> retrieveBPartnerLocationsInTrx(final BPartnerId bpartnerId)
```

Comment:
do we need this as a public method? I ask bycause I'm a bit reluctant to add new methods to our API that return "raw" datarecords..particularly if we don't need them

Reply:
No. We don't need it as public. Made it private.